packages/ui/src/overview/components/DeadlineList/DeadlineList.tsx

```diff
@@ -19,14 +21,63 @@ export interface DeadlineListProps {
   workingGroup?: WorkingGroup
 }
 
+export type DeadlineNamespace = 'election' | 'proposals' | 'openings' | 'upcomingOpenings'
+
+const initializer: Record<DeadlineNamespace, string[]> = {
+  proposals: [],
+  openings: [],
+  upcomingOpenings: [],
+  election: [],
+}
+
 export const DeadlineList: React.FC<DeadlineListProps> = React.memo(({ workingGroup }) => {
   const { t } = useTranslation('overview')
-  const { isLoading: proposalLoading, proposals } = useProposals({ status: 'active', filters: { stage: 'deciding' } })
-  const { election } = useCurrentElection()
-  const { isLoading: upcomingLoading, upcomingOpenings } = useUpcomingOpenings({ groupId: workingGroup?.id })
-  const { isLoading: openingLoading, openings } = useOpenings({ groupId: workingGroup?.id, type: 'open' })
+  const { election, proposals, upcomingOpenings, openings, isLoading } = useDeadlines({ groupId: workingGroup?.id })
+  const [storageDeadlines, setStorageDeadlines] = useLocalStorage<Record<DeadlineNamespace, string[]>>('deadlines')
+  const [tempDeadlines, setTempDeadlines] = useState<Record<DeadlineNamespace, string[]> | null>(null)
+  const tempRef = useRef<boolean>(false)
+  useEffect(() => {
+    if (!tempRef.current && !tempDeadlines) {
+      setTempDeadlines(storageDeadlines ?? initializer)
+      tempRef.current = true
+    }
+
+    return () => {
+      if (tempDeadlines) {
+        setStorageDeadlines(tempDeadlines)
+      }
+    }
+  }, [storageDeadlines, tempDeadlines])
+
+  const helper = (namespace: DeadlineNamespace) => (id: string) => {
+    setTempDeadlines((prev) => {
+      return prev
+        ? {
+            ...prev,
+            [namespace]: [...prev[namespace], id],
+          }
+        : null
+    })
+  }
+
+  const deadlines = useMemo((): Record<DeadlineNamespace, any[]> => {
+    return {
+      proposals: tempDeadlines?.proposals
+        ? proposals.filter((proposal) => !tempDeadlines.proposals.includes(proposal.id))
+        : proposals,
+      election: tempDeadlines?.election
+        ? election?.candidates.filter((candidate) => !tempDeadlines.election.includes(candidate.id)) ?? []
+        : election?.candidates ?? [],
+      openings: tempDeadlines?.openings
+        ? openings.filter((opening) => !tempDeadlines.openings.includes(opening.id))
+        : openings,
+      upcomingOpenings: tempDeadlines?.upcomingOpenings
+        ? upcomingOpenings.filter((upcomingOpening) => !tempDeadlines.upcomingOpenings.includes(upcomingOpening.id))
+        : upcomingOpenings,
+    }
+  }, [storageDeadlines?.proposals, proposals, election, upcomingOpenings, openings])
```

Comment:
It's great that the deadline are all queried in one the hook, but also I think all the filtering/(maybe later sorting) should happen in the same hook too